### ISSUE TITLE
EnumsShouldHavePluralNames (CA1717) reports on singular names

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#950]
#### Analyzer package

Microsoft.ApiDesignGuidelines.Analyzers
#### Analyzer

CA1717: EnumsShouldHavePluralNames
#### Repro steps

``` csharp
public enum ClockSynchronizationStatus { }
public enum NetworkAddress { }
```
#### Expected behavior

No report on the names above, as they are singular form.
#### Actual behavior

Both enums are reported by CA1717.

Looking at the source, pluralization is determined by:

``` csharp
        public static bool IsPlural(this string word)
        {
            if (!word.EndsWith("s", StringComparison.OrdinalIgnoreCase) &&
                !word.EndsWith("i", StringComparison.OrdinalIgnoreCase) &&
                !word.EndsWith("ae", StringComparison.OrdinalIgnoreCase))
            {
                return false;
            }
            return true;
        }
```

which seems too simplistic. Perhaps a library as suggested in http://stackoverflow.com/questions/475705/pluralize-singularize could help to improve acuracy here.

**Comment by srivatsn** [comment]
What does the FxCop implementation do for these cases?

**Comment by jinujoseph** [comment]
The plural check is definitely not enough and that's the reason I was looking at pluralizationservice (https://msdn.microsoft.com/en-us/library/system.data.entity.design.pluralizationservices.pluralizationservice(v=vs.110).aspx ) but since it's a portal lib , can't use the same. So currently doing exactly how Fxcop implemented. 
http://index/#NamingRules/FlagsEnumsShouldHavePluralNames.cs

**Comment by jinujoseph** [comment]
pending humanizer approval

**Comment by jinujoseph** [comment]
also reported [here](https://developercommunity.visualstudio.com/content/problem/307009/ca1717-only-flagsattribute-enums-should-have-plura.html)
